documentation/prometheus-mixin/alerts.libsonnet

```diff
@@ -259,6 +259,82 @@
               description: 'Prometheus %(prometheusName)s has missed {{ printf "%%.0f" $value }} rule group evaluations in the last 5m.' % $._config,
             },
           },
+
+          {
+            alert: 'PrometheusScrapeSeriesCardinality',
+            expr: |||
+              sum(scrape_series_added) by (job) > 100 * 1000
+            ||| % $._config,
+            'for': '30m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              summary: 'Prometheus scrape series cardinality is exceeding server abilities.',
+              description: 'Prometheus %(prometheusName)s has {{ printf "%%.0f" $value }} scrape series for {{$labels.job}} job in last 30m.' % $._config,
+            },
+          },
+          {
+            alert: 'PrometheusScrapeSamplesCardinality',
+            expr: |||
+              sum(scrape_samples_scraped) by (job) > 100 * 1000
+            ||| % $._config,
+            'for': '30m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              summary: 'Prometheus scrape samples cardinality is exceeding server abilities.',
+              description: 'Prometheus %(prometheusName)s has {{ printf "%%.0f" $value }} scrape samples for {{$labels.job}} job in last 30m.' % $._config,
+            },
+          },
+          {
+            alert: 'PrometheusScrapeSeriesExplodingCardinality',
+            expr: |||
+              (
+                deriv(scrape_series_added[1h]) > 3 * 0.1
```

Comment:
Thinking about this rule some, right now this would alert if any instance being scraped is adding series which may or may not be bad. For example, kube state metrics rolling would create a huge amount of metrics once, which may well send an alert for this even though it is stable afterward.

Would an alert for exploding cardinality be better looking at something like the increase of `prometheus_tsdb_head_series_created_total`?

Reply:
I see two problems with using increase of `prometheus_tsdb_head_series_created_total` for this alert:
- prometheus startup causes very high increase of this metric
- larger k8s clusters with high rotation of pods also cause high increase